pkg/tools/multiexecutor/executor_func.go

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2020-2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multiexecutor
+
+// ExecutorFunc - function type, simulates serialize.Executor behavior
+//		Can be used to guarantee exclusive, in order execution of functions.
+type ExecutorFunc func(f func()) <-chan struct{}
+
+// AsyncExec - simulates serialize.Executor AsyncExec call
+// 		Guarantees f() will be executed Exclusively and in the Order submitted.
+//		It immediately returns a channel that will be closed when f() has completed execution.
+func (e ExecutorFunc) AsyncExec(f func()) <-chan struct{} {
+	return e(f)
+}
```

Comment:
do we need this as a public?

Reply:
Yes. At the moment it become a public tool ExecutorFunc has to be also public

Reply:
See this https://github.com/networkservicemesh/sdk/pull/700#discussion_r573101131

Reply:
Fixed by https://github.com/networkservicemesh/sdk/pull/703/commits/8635adba8e8e0d05eea0f4d889567564f62dd036